addons/binding/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/internal/serial/ValloxSerialInterface.java

```diff
@@ -0,0 +1,453 @@
+    }
+
+    /**
+     * Start a heartbeat thread that regularly polls a vallox variable and
+     * automatically reconnects if it fails.
+     */
+    public void startHeartbeat() {
+        heartbeatExecutor = Executors.newSingleThreadExecutor();
+        heartbeatExecutor.submit(() -> {
+            while (!shutdownListening) {
+                try {
+                    Thread.sleep(MINUTE);
+                    sendPoll(ValloxProperty.SelectStatus);
+                } catch (Exception e) {
+                    String msg = "Exception sending heartbeat poll: " + e.toString();
+                    logger.error(msg);
+                    for (StatusChangeListener l : this.getStatusListener()) {
+                        l.statusChanged(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, msg);
+                    }
+                    try {
+                        this.reconnect();
+                    } catch (IOException e1) {
+                        logger.error("Exception reconnecting vallox: " + e.toString());
+                    }
+                }
+            }
+        });
+        logger.info("Start heartbeat check to vallox!");
+    }
+
+    /**
+     * Read one telegram.
+     *
+     * @return the received telegram or null if no telegram could be read
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    private Telegram receive() throws IOException, InterruptedException {
+        int available = inputStream.available();
+        if (available < ValloxProtocol.LENGTH) {
+            Thread.sleep(200);
+            return null;
+        }
+        int domain = inputStream.read();
+        if (domain != ValloxProtocol.DOMAIN) {
+            logger.warn("Received Telegram has no Domain byte, ignoring telegram: {}",
+                    Telegram.byteToHex((byte) domain));
+            return null;
+        }
+
+        // now we're happy -> read the telegram
+        int sender = inputStream.read();
+        int receiver = inputStream.read();
+        int command = inputStream.read();
+        int arg = inputStream.read();
+        int checksum = inputStream.read();
+        int computedChecksum = (domain + sender + receiver + command + arg) & 0x00ff;
+
+        if (checksum != computedChecksum) {
+            logger.warn("Received Telegram has invalid checksum, ignoring telegram.");
+            return null;
+        }
+
+        // only read telegrams that are for us
+        if (receiver == receiverID || receiver == senderID || receiver == ValloxProtocol.ADDRESS_PANELS) {
+            return new Telegram((byte) sender, (byte) receiver, (byte) command, (byte) arg);
+        }
+        logger.debug("Ignoring telegram not for us: {} {} {} {}", sender, receiver, command, arg);
+        return null;
+    }
+
+    public void stopListening() {
+        if (listenerExecutor != null) {
+            try {
+                logger.info("attempt to shutdown listener");
```

Comment:
Please change the logging level to `debug`. `trace` can be used for even more details, where necessary.